packages/kit/src/runtime/server/page/render.js

```diff
@@ -136,7 +136,7 @@ export async function render_response({
 					page: {
 						host: ${page && page.host ? s(page.host) : 'location.host'}, // TODO this is redundant
 						path: ${s(page && page.path)},
-						query: new URLSearchParams(${page ? s(page.query.toString()) : ''}),
+						query: new URLSearchParams(document.location.search),
```

Comment:
I think I read somewhere from Rich that we don't really want this, I forgot where I saw it but it's something about different results in server and client. Aside from that, can we also use `document` here?

Reply:
It was in https://github.com/sveltejs/kit/issues/669 or one of the linked issues. The issue was that folks wanted to set it in this way so that they could use it on pre-rendered pages, but then it'd only work on the client and the pre-rendered pages wouldn't match. However, it's okay to do as part of this PR because we're explicitly disallowing it from being used during pre-rendering so no one can get unexpected results.

I had to change it here because if I try to access `page.query` as the code was doing before it'd throw the validation error, which I could have possibly gotten around in other ways, but at the cost of messier code

Reply:
This would in fact break an app I recently worked on, in which we were using SvelteKit to generate HTML embeds and using the page query to configure them. Those embeds were then placed in a context where `location.search` would have equalled something incorrect.

While it's something of a niche situation, I'd prefer that we keep the ability to generate HTML that behaves in a predictable way. Couldn't we simply address this TODO instead? https://github.com/sveltejs/kit/blob/aad27f4ed3a2a57f68c6cb55076b9d3132bcbaac/packages/kit/src/core/adapt/prerender.js#L268-L270

Reply:
I'm not quite understanding how `page.query` and `location.search` would be different or exactly what you're doing, but I've reverted the change to this line, so hopefully that addresses your concern?